Create a 5x4 matrix: 5 accounts × 4 asset types (Stock, Bond, ETF, Crypto)
Values represent holdings in €1000s

Questions:
1. Create the matrix with random values between 10 and 100.
2. Calculate total holdings per account (row sums).
3. Calculate total holdings per asset type (column sums).
4. Find which account has the highest total holdings.

In [1]:
import numpy as np

Question 1

In [18]:
# 10 and 100 passed as the first and second arguments
# indicate that the random integers should be between
# 10 and 100 (both included).
holdings = np.random.randint(10, 100, size=(5, 4))

In [19]:
print('holdings:', holdings)

holdings: [[24 93 72 61]
 [71 23 21 97]
 [92 80 10 96]
 [77 92 81 68]
 [35 61 11 94]]


Question 2

In [30]:
# The axis=1 keyword argument specifies that the sum of
# each row should be computed.
account_totals = np.sum(holdings, axis=1)

In [31]:
print('account_totals:', account_totals)

account_totals: [250 212 278 318 201]


Question 3

In [22]:
# The axis=0 keyword argument specifies that the sum of
# each column should be computed.
asset_totals = np.sum(holdings, axis=0)

In [29]:
print('asset_totals:', asset_totals)

asset_totals: [299 349 195 416]


Question 4

In [27]:
# argmax() returns the index of the maximum value.
max_account_index = np.argmax(account_totals)

In [28]:
print('max_account_index:', max_account_index)

max_account_index: 3


In [32]:
max_holdings = account_totals[max_account_index]

318 is the sum for the fourth row.

Keep in mind it is also the row with the maximum sum.

In [33]:
print('max_holdings:', max_holdings)

max_holdings: 318
